dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -434,23 +436,34 @@ public Workflow getWorkflow(@ApiParam(hidden = true) @Parameter(hidden = true, n
     @GET
     @Path("/{workflowId}/workflowVersions")
     @UnitOfWork
-    @ApiOperation(nickname = "getWorkflowVersions", value = "Return first 200 versions in an entry", authorizations = {
+    @ApiOperation(nickname = "getWorkflowVersions", value = "Return paginated versions in an entry", authorizations = {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME)}, response = WorkflowVersion.class, responseContainer = "List")
-    @Operation(operationId = "getWorkflowVersions", description = "Return first 200 versions in an entry", security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME))
+    @Operation(operationId = "getWorkflowVersions", description = "Return paginated versions in an entry", security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME))
     @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "Get list workflow versions in a workflow", content = @Content(
         mediaType = "application/json",
         array = @ArraySchema(schema = @Schema(implementation = WorkflowVersion.class))))
     @ApiResponse(responseCode = HttpStatus.SC_BAD_REQUEST + "", description = "Bad Request")
-    public Set<WorkflowVersion> getWorkflowVersions(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user") @Auth User user,
-        @ApiParam(value = "workflowID", required = true) @Parameter(name = "workflowId", description = "id of the worflow", required = true, in = ParameterIn.PATH) @PathParam("workflowId") Long workflowId) {
+    public Set<WorkflowVersion> getWorkflowVersions(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user") @Auth Optional<User> user,
+        @ApiParam(value = "workflowID", required = true) @Parameter(
+                name = "workflowId", description = "id of the worflow", required = true, in = ParameterIn.PATH) @PathParam("workflowId") Long workflowId,
+        @QueryParam("limit")  @DefaultValue("200") Integer limit,
+        @QueryParam("offset") @DefaultValue("0") Integer offset,
+        @Context HttpServletResponse response) {
         Workflow workflow = workflowDAO.findById(workflowId);
         checkNotNullEntry(workflow);
-        checkCanExamine(user, workflow);
 
-        List<WorkflowVersion> versions = this.workflowVersionDAO.getWorkflowVersionsByWorkflowId(workflow.getId(), VERSION_PAGINATION_LIMIT, 0);
+        if (user.isEmpty() && !workflow.getIsPublished()) {
+            LOG.error("Unpublished workflow requires authentication");
+            throw new CustomWebApplicationException("Unpublished workflow requires authentication", HttpStatus.SC_BAD_REQUEST);
+        }
```

Comment:
Can use `checkCanRead` which checks if it's published or if the user has access to it and throws an error if not:
https://github.com/dockstore/dockstore/blob/cec35ec73690bc7af06daed764d4c86687246591/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java#L91-L99

Reply:
Related comment:
Looks like previously, this method called `checkCanExamine`, which confirms the access necessary to see fields that only owner-type users should be able to see.  If we switch to confirming access via `checkCanRead` or similar, this implies that the endpoint is changing so that it provides access to all users, and perhaps, in certain situations, we'd need to filter out things that everyone shouldn't be able to see, like hidden versions.

Reply:
`checkCanRead` looks to be sufficient as it returns true if the entry is either published or the `canExamine` function. https://github.com/dockstore/dockstore/blob/081a733abbdc5a0f8cc0d75ece805cf710153ea0/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java#L209

Reply:
Yes, `checkCanRead` works fine, but if you use it, the endpoint will return Versions to _anyone_ if the entry is published.  If that's a change from how the endpoint worked before (and it appears like you used to _have_ to authenticate and have owner-type access to the entry), we might need to make sure that it doesn't show versions that shouldn't be public, like those that are hidden.  (I think.)

Reply:
Thank you for your comment. I added changes to remove hidden versions from being seen to those without access